tests/hierarchical/hierarchical_implicit_reduce.cpp

```diff
@@ -8,15 +8,23 @@
 
 #include "../common/common.h"
 
-#define TEST_NAME hierarchical_reduce
+#define TEST_NAME hierarchical_implicit_reduce
 
 namespace TEST_NAMESPACE {
 
-static const int groupItems1d = 2;
+static const int globalItems1d = 2;
+static const int globalItems2d = 2;
+static const int globalItems3d = 2;
 static const int localItems1d = 2;
-static const int groupItemsTotal = groupItems1d * groupItems1d * groupItems1d;
-static const int localItemsTotal = localItems1d * localItems1d * localItems1d;
-static const int numGroups = groupItemsTotal / localItemsTotal;
+static const int localItems2d = 1;
+static const int localItems3d = 1;
+static const int groupItems1d = (globalItems1d / localItems1d);
+static const int groupItems2d = (globalItems2d / localItems2d);
+static const int groupItems3d = (globalItems3d / localItems3d);
+static const int globalItemsTotal =
+    (globalItems1d * globalItems2d * globalItems3d);
+static const int localItemsTotal = (localItems1d * localItems2d * localItems3d);
+static const int numGroups = (globalItemsTotal / localItemsTotal);
 
 static const int inputSize = 32;
```

Comment:
I guess that most of these could be constexpr.
Otherwise it would be nice to have some less trivial sizes to detect some trivial bugs, such as wrong dimensions, transpositions, etc.
For example use some different small prime numbers.

Reply:
Done and done, they're all prime numbers now - global items are a product of two prime numbers.